SimplenoteUITests/UIDs.swift

```diff
@@ -32,11 +33,13 @@ enum UID {
 		static let noteEditorChecklist = "Inserts a new Checklist Item"
 		static let noteEditorInformation = "Information"
 		static let noteEditorMenu = "note-menu"
-		static let noteCellTrash = "icon trash"
+		static let itemTrash = "icon trash"
```

Comment:
You mention this PR being too long, still. I don't think that's the case it's a decent size 👍 

One thing I'd recommend, though, would be making changes like renaming a value in their own commit.

I also noticed you listed 3 operations that the tests verify in the commit message. Those, too, could have been dedicated commits.

- Commit 1: Rename noteCellTrash to itemTrash in UI tests UIDs
- Commit 2: Add UI test to add a tag to a note
- Commit 3: Add UI test to delete all tags from the sidebar
- Commit 4: Add UI test to select a tag in the sidebar

The value of having small commits is that it's easier to look at the changes in isolation. Also, if one small commit introduces an error, it's easier to revert that without loosing the rest of the work done in the PR.